fix(language_server): use the first Span of the message as the primary Diagnostic range

Sysix opened the pull request:
This PR aligns the main label/span with `oxlint`.
`oxlint` will use the first available span as the error. The LSP should use this span for the red line.

```
let a: {
  b?: {
    c: number;
  };
} = {};

for(var i = 0; i < 10; i--){}
console.log(a?.b?.c!);
```

Note that `file.ts:7:16` means the line/column number and is linkable.

```
  ⚠ eslint(for-direction): The update clause in this loop moves the variable in the wrong direction
   ╭─[for_direction.ts:7:16]
 6 │ 
 7 │ for(var i = 0; i < 10; i--){}
   ·                ───┬──  ─┬─
   ·                   │     ╰── with this update
   ·                   ╰── This test moves in the wrong direction
 8 │ console.log(a?.b?.c!);
   ╰────
  help: Use while loop for intended infinite loop

  ⚠ typescript-eslint(no-non-null-asserted-optional-chain): Optional chain expressions can return undefined by design: using a non-null assertion is unsafe and wrong.
   ╭─[for_direction.ts:8:20]
 7 │ for(var i = 0; i < 10; i--){}
 8 │ console.log(a?.b?.c!);
   ·                 ┬  ┬
   ·                 │  ╰── non-null assertion made after optional chain
   ·                 ╰── optional chain used
 9 │ 
   ╰────
  help: Remove the non-null assertion.
```

https://github.com/oxc-project/oxc/blob/main/crates/oxc_language_server/fixtures/linter/issue_9958/issue.ts

Main:
<img width="258" height="88" alt="grafik" src="https://github.com/user-attachments/assets/7da86f38-9789-4e04-b08b-18aa783a93ed" />

This PR:
<img width="265" height="108" alt="grafik" src="https://github.com/user-attachments/assets/2a031f84-04c3-49a8-8409-a3b1822c55a5" />

camc314 commented:
### Merge activity

* **Oct 8, 10:11 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
fix(language_server): use the first Span of the message as the primary Diagnostic range (#14057)

This PR aligns the main label/span with `oxlint`.
`oxlint` will use the first available span as the error. The LSP should use this span for the red line.

```
let a: {
  b?: {
    c: number;
  };
} = {};

for(var i = 0; i < 10; i--){}
console.log(a?.b?.c!);
```

Note that `file.ts:7:16` means the line/column number and is linkable.

```
  ⚠ eslint(for-direction): The update clause in this loop moves the variable in the wrong direction
   ╭─[for_direction.ts:7:16]
 6 │
 7 │ for(var i = 0; i < 10; i--){}
   ·                ───┬──  ─┬─
   ·                   │     ╰── with this update
   ·                   ╰── This test moves in the wrong direction
 8 │ console.log(a?.b?.c!);
   ╰────
  help: Use while loop for intended infinite loop

  ⚠ typescript-eslint(no-non-null-asserted-optional-chain): Optional chain expressions can return undefined by design: using a non-null assertion is unsafe and wrong.
   ╭─[for_direction.ts:8:20]
 7 │ for(var i = 0; i < 10; i--){}
 8 │ console.log(a?.b?.c!);
   ·                 ┬  ┬
   ·                 │  ╰── non-null assertion made after optional chain
   ·                 ╰── optional chain used
 9 │
   ╰────
  help: Remove the non-null assertion.
```

https://github.com/oxc-project/oxc/blob/main/crates/oxc_language_server/fixtures/linter/issue_9958/issue.ts

Main:
<img width="258" height="88" alt="grafik" src="https://github.com/user-attachments/assets/7da86f38-9789-4e04-b08b-18aa783a93ed" />

This PR:
<img width="265" height="108" alt="grafik" src="https://github.com/user-attachments/assets/2a031f84-04c3-49a8-8409-a3b1822c55a5" />

camc314 commented on crates/oxc_language_server/src/linter/error_with_position.rs:
```diff
@@ -71,24 +64,14 @@ fn message_with_position_to_lsp_diagnostic(
 
-    let range = related_information.as_ref().map_or(
+    let range = message.labels.as_ref().map_or(Range::default(), |labels| {
+        let start = labels.first().map(SpanPositionMessage::start).cloned().unwrap_or_default();
+        let end = labels.first().map(SpanPositionMessage::end).cloned().unwrap_or_default();
```
oh these aren't spans, and don't have a `primary()`, method on them.
